components/dada-execute/src/step/traversal.rs

```diff
@@ -240,6 +242,15 @@ impl Stepper<'_> {
                     ))
                 }
             }
+            ObjectData::Tuple(tuple) => {
+                let field_name_str = field_name.as_str(self.db);
```

Comment:
Nit: Feels like it might be better to parse the field name once into an integer...? Not that this code is mega optimized. :)

Reply:
Oh, I see from your tests, maybe we don't want `foo.01` to work... hmm.... interesting question. :)